week3project.py

```diff
@@ -0,0 +1,77 @@
+import zipfile2 as zipfile
+
+from PIL import Image
+import pytesseract
+import cv2 as cv
+import numpy as np
+
+face_cascade = cv.CascadeClassifier('haarcascade_frontalface_default.xml')
+
+# function that finds the faces and creates bounding boxes
+def find_and_crop(image_file):
+    ar_image = np.array(image_file)[:,:,::-1]
+    ar_image2 = cv.cvtColor(ar_image, cv.COLOR_BGR2GRAY)
+    faces = face_cascade.detectMultiScale(ar_image2, 1.3, minNeighbors = 4, minSize = (25, 25), maxSize = (270, 270), flags = cv.CASCADE_SCALE_IMAGE)
+    return croptheface(image_file, faces)
+
+# function that crops the faces out of the original image using bounding boxes
+def croptheface(filename, faces):
+    list_of_faces = []
+    for x, y, w, h in faces:
+        list_of_faces.append(filename.crop((x, y, x + w, y + h)))
+    return list_of_faces
+
+def thumbs_n_nails(pic):
+    pic1 = pic.copy()
+    pic1.thumbnail((100,100))
+    return pic1
+
+# input that will be used later
+keyword = input('please enter a word you wish to find on a newspaper page, which will then output a contact sheet of cropped face images on the same page: ')
+# creating a dictionary to store PIL.Image files, text strings and bounding boxes in
+global_dict = {}
+# accum through the list of file names in zip archive
+accum = 0
+# Unpacking the zip archive and saving PIL.Images into global_dict
+with zipfile.ZipFile('small_img.zip', 'r') as archive:
+    for image in archive.infolist():
+        one_zip_img = archive.open(image)
+        global_dict[f'{archive.namelist()[accum]}']= {}
+        global_dict[f'{archive.namelist()[accum]}']['PIL'] = Image.open(one_zip_img)
+        accum += 1
+
+
+# Main code for the project - to loop through the keys in global_dict (they are the same as file names)
+for page in global_dict:
```

Comment:
You can iterate over keys and values of the dictionary in the following manner:
`for page, image in global_dict.items():`